std/typecons.d

```diff
@@ -1168,6 +1169,99 @@ template Tuple(Specs...)
 }
 
 /**
+    Allows to unpack a $(D Tuple) range to arguments when iterating using $(D foreach).
+
+    Examples:
+    ---
+    foreach (string name, int id; [tuple("john", 5), tuple("jeff", 2)].unpackIterator)
+    {
+        writefln("name: %s, id: %s" name, id);
+    }
+    ---
+
+    Examples:
+    ---
+    alias T = Tuple!(string,string);
+    T[] array = [ T("a", "b"), T("c", "d") ];
+    foreach (size_t index, string first, string second; array.unpackIterator)
+    {
+        writefln("index: %s, first: %s, second: %s", index, first, second);
+    }
+    ---
+*/
+auto unpackIterator(Range)(Range r)
+    if (isInputRange!(Unqual!Range) && isTuple!(ElementType!Range))
+{
+    static struct Result
+    {
+        private alias R = Unqual!Range;
+        private alias Types = ElementType!R.Types;
+        public R source;
+
+        int opApply(scope int delegate(Types args) dg)
+        {
+            while (!source.empty)
+            {
+                const int result = dg(source.front.expand);
+                source.popFront();
+
+                if (result)
+                    return result;
+            }
+            return 0;
+	    }
+
+        int opApply(scope int delegate(size_t, Types) dg)
+        {
+            size_t i = 0;
+            while (!source.empty)
+            {
+                const int result = dg(i, source.front.expand);
+                source.popFront();
+                i ++;
+
+                if (result)
+                    return result;
+            }
+            return 0;
+	    }
+    }
+
+    return Result(r);
+}
+
+unittest
```

Comment:
For safe, I'd have to add `@safe` to the delegate in `opApply`, which seems to be limiting functionality. Since I don't really know `@safe`, I can't argue on this point. So you say I should enforce the foreach body to be `@safe`?